typescript/cli/src/config/warp.ts

```diff
@@ -134,15 +140,24 @@ export async function createWarpRouteDeployConfig({
     const isNft =
       type === TokenType.syntheticUri || type === TokenType.collateralUri;
 
-    const mailbox = await detectAndConfirmOrPrompt(
-      async () => {
-        const addresses = await context.registry.getChainAddresses(chain);
-        return addresses?.mailbox;
-      },
-      `For ${chain}, enter the`,
-      'mailbox address',
-      'hyperlane-registry',
-    );
+    // default to the mailbox from the registry and if not found ask to the user to submit one
+    const chainAddresses = await context.registry.getChainAddresses(chain);
+
+    const chainMetadata = await context.registry.getChainMetadata(chain);
+    if (!chainMetadata) {
+      throw new Error(`Metadata not found for chain "${chain}"`);
+    }
+
+    let mailbox: Address;
+    if (chainAddresses) {
+      mailbox = chainAddresses.mailbox;
+    } else {
+      mailbox = await input({
```

Comment:
I think we prefer `const` declarations when possible
```suggestion
    const mailbox = chainAddresses?.mailbox ?? await input({
```
can see the argument that the if/else is more readable